app/models/invenio_rdm_record_converter.rb

```diff
@@ -96,23 +96,27 @@ def owner_info
     user = User.find_by(username: @generic_file.depositor)
 
     if user
-      {
-        "netid": user.username,
-        "email": user.email
-      }
+      { user.username => user.email }
     else
-      {
-        "netid": "unknown",
-        "email": "unknown"
-      }
+      { "unknown": "unknown" }
     end
   end
 
   def file_permissions
-    permission_data = Hash.new { |h,k| h[k] = [] }
+    permission_data = Hash.new
+
+    permission_data["owner"] = owner_info
 
     @generic_file.permissions.each do |permission|
-      permission_data[permission.access] << permission.agent_name
+      permission_data[permission.access] ||= Hash.new
+
+      if @role_store[permission.agent_name]
+        permission_data[permission.access].merge!(@role_store[permission.agent_name])
+      elsif user = User.find_by(username: permission.agent_name)
```

Comment:
If this proves too slow, we can probably preload the user in memory since there is only hundreds of them or so.